02-microarray/differential-expression_microarray_01.Rmd

```diff
@@ -0,0 +1,349 @@
+
+```{r}
+metadata <- metadata %>%
+  dplyr::rename("genotype" = `genotype/variation`) # This step will not be the same (or might not be needed at all) with a different dataset
+```
+
+Now we will create a model matrix based on our newly renamed `genotype` variable. 
+
+```{r}
+# Create the design matrix from the genotype information
+des_mat <- model.matrix(~ metadata$genotype)
+```
+
+## Apply linear model
+
+After applying our data to linear model, in this example we apply empirical Bayes smoothing and Benjamini-Hochberg multiple testing correction. 
+The `topTable()` function default is to use Benjamini-Hochberg but this can be changed to a different method using the `adjust.method` argument.
+
+```{r}
+# Apply linear model to data
+fit <- lmFit(df, design = des_mat)
+
+# Apply empirical Bayes to smooth standard errors
+fit <- eBayes(fit)
+
+# Apply multiple testing correction and obtain stats
+stats_df <- topTable(fit, number = nrow(df)) %>%
+  tibble::rownames_to_column("Gene")
+```
+
+Let's take a peek at what our results table looks like. 
+
+```{r}
+stats_df
+```
+
+By default, results are ordered by largest `B` (the log odds value) to the smallest, which means your most differentially expressed genes should be toward the top. 
+
+See the help page by using `?topTable` for more information and options for this table. 
+
+## Check results by plotting one gene
+
+To test if these results make sense, we can make a plot of one of top genes. 
+Let's try `ENSDARG00000104315`. 
+
+```{r}
+top_gene_df <- df %>%
+  # Extract this gene from `df`
+  dplyr::filter(rownames(.) == "ENSDARG00000104315") %>%
+  # Transpose so the gene is a column
+  t() %>%
+  # Transpose made this a matrix, let's make it back into a data.frame like before
+  data.frame() %>%
+  # Store the sample ids as their own column instead of being row names
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Join on the selected columns from metadata
+  dplyr::inner_join(dplyr::select(
+    metadata,
+    refinebio_accession_code,
+    genotype
+  ))
+```
+
+```{r}
+ggplot(top_gene_df, aes(x = genotype, y = ENSDARG00000104315, color = genotype)) +
+  geom_jitter(width = 0.2) + # We'll make this a jitter plot
+  theme_classic() # This makes some aesthetic changes
+```
+
+These results make sense.
+The overexpressing CREB group samples are have much higher expression values for ENSDARG00000104315 than the control samples do. 
+
+
+## Write results to file
+
+This will be saved to our results directory. 
+
+```{r}
+readr::write_tsv(stats_df, file.path(
+  results_dir,
+  "GSE71270_limma_results.tsv" # Replace with a relevant output name
+))
+```
+
+## Make a volcano plot
+
+We'll use the `EnhancedVolcano` package's main function to plot our data [@Zhu2018].
+
+```{r}
+volcano_plot <- EnhancedVolcano::EnhancedVolcano(stats_df,
+  lab = "Gene",
+  x = "logFC",
+  y = "adj.P.Val"
+)
+```
+
```

Comment:
We should probably add a sentence here commenting on the possible interpretation of the plot above as well.

Reply:
I've added some more guidance about `EnhancedVolcano` plot! Good call.